### PR TITLE
Include human readable responses in response parsing tests

### PR DESCRIPTION
**Pull request by cbuescher**
As a follow up to #22649, this changes the resent tests for parsing parts of search responses to randomly set the humanReadable() flag of
the XContentBuilder that is used to render the responses. This should help to test that we can parse back thoses classes if the user specifies `?human=true` in the request url.

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM, I take it that you didn't find any problems when adding this new check? @tlrx want to have a quick look too?

**Comment by cbuescher** [comment]
No, all good, I think the human flag only makes a difference in the Profile section output at the moment. These tests are mostly added to protect us from suprises in the future.

File: core/src/test/java/org/elasticsearch/action/delete/DeleteResponseTests.java [hunk 1]
```diff
@@ -92,8 +93,8 @@ private static DeleteResponse randomDeleteResponse() {
         ShardId shardId = new ShardId(randomAsciiOfLength(5), randomAsciiOfLength(5), randomIntBetween(0, 5));
         String type = randomAsciiOfLength(5);
         String id = randomAsciiOfLength(5);
-        long seqNo = randomIntBetween(-2, 5);
-        long version = (long) randomIntBetween(0, 5);
+        long seqNo = randomBoolean() ? randomNonNegativeLong() : randomIntBetween(0, 10000);
```
Comment (tlrx) [inline core/src/test/java/org/elasticsearch/action/delete/DeleteResponseTests.java]: Can we use 
`randomFrom(SequenceNumbersService.UNASSIGNED_SEQ_NO, randomNonNegativeLong(), randomIntBetween(0, 10000)) `

instead?

File: core/src/test/java/org/elasticsearch/action/index/IndexResponseTests.java [hunk 2]
```diff
@@ -165,8 +167,8 @@ private static IndexResponse randomIndexResponse() {
         ShardId shardId = new ShardId(randomAsciiOfLength(5), randomAsciiOfLength(5), randomIntBetween(0, 5));
         String type = randomAsciiOfLength(5);
         String id = randomAsciiOfLength(5);
-        long seqNo = randomIntBetween(-2, 5);
-        long version = (long) randomIntBetween(0, 5);
+        long seqNo = randomBoolean() ? randomNonNegativeLong() : randomIntBetween(0, 10000);
```
Comment (tlrx) [inline core/src/test/java/org/elasticsearch/action/index/IndexResponseTests.java]: Same here

**Review by tlrx — APPROVED** [review]
LGTM, left two minor comments

**Comment by cbuescher** [comment]
Added to 5.x with 25a219f0c9368a5f15e9b6fe96cfea3f8e57e886

**Pull request merged**